### DISCUSSION TITLE
How do I fetch a bunch of entities based of a many-to-many relationship?

### DISCUSSION THREAD

**Discussion by cravall** [SeaQL/sea-orm#1998] [Q&A]
EDIT: I actually don't have a many to many relationship like the cake to filling as described... it's more of a linking situation. I'll open another question because this one isn't right.

~I'm ~looking to fetch a bunch of items based off a many-to-many relationship. If we take the example from here: https://www.sea-ql.org/SeaORM/docs/relation/many-to-many/~

~How would we then create a function that got all the Fillings without getting the Cake first (e.g. not using `with_related` after loading the cake from the database) when we just have the cake id?~

~I think I can do something like:~
```
        let dummy_cake = cake::Model {
            id: cake_id,
            // .. all the other fields
        };
        
        let fillings = dummy_cake
            .find_related(filling::Entity)
            .all(&db)
            .await?;
```

~But that kinda feels icky to create the dummy_cake , especially if there's a bunch of fields that I just don't need for the query.~

~I'm just making up air code, but is there something like this available?~
```
let filings = cake::Entity::find_related(filling:Entity).filter(cake::Id.eq(cake_id)).all(&db).await?;
```
~Thanks!~

~Ps. I also posted this in the discord... just wasn't sure the best place was to get support :-)~

**Comment by cosmincatalin** [comment]
If you know the cake instance, it means you know its ID, so then you just need to query the fillings and filter based on the cake FK. Does this make sense?

**Reply by cravall** [reply]
Hi @cosmincatalin,

It makes sense, as in I understand what you are saying... but I don't understand how to translate that into SeaOrm code.

Can I use the `filling::Entity::.find().join(stuff_in_here)` with a many-to-many relationship? Or is there a way to join one table and then join a second table based off the first joined table?

**Reply by cravall** [reply]
Sorry @cosmincatalin ... I actually came back to this and the problem isn't as I initially explained. Your comment does make sense in the context of my original question.

I'll open another question to better explain my problem.

**Comment by cravall** [comment]
Closing because the problem is actually different to how I have explained it.
